frappe/utils/typst.py

```diff
@@ -0,0 +1,103 @@
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from uuid import uuid4
+
+from pypdf import PdfReader, PdfWriter
+
+import frappe
+from frappe.core.doctype.file.utils import get_local_image, get_web_image
+from frappe.model.document import Document
+from frappe.utils.data import to_markdown
+from frappe.utils.pdf import get_file_data_from_writer
+
+
+def print_with_typst(
+	doctype, name, print_format, as_pdf, doc, output: PdfWriter | None = None, password=None
+):
+	if not doc:
+		doc = frappe.get_doc(doctype, name)
+
+	template = frappe.db.get_value("Print Format", print_format, "raw_commands")
+	temp_dirname = Path("/tmp") / str(uuid4())
+	temp_dirname.mkdir(parents=True, exist_ok=False)
```

Comment:
You likely won't have permissions to do this in many production environments - https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile